code/go/0chain.net/smartcontract/dbs/event/stakepool.go

```diff
@@ -13,60 +14,31 @@ import (
 )
 
 type providerRewardsDelegates struct {
-	rewards       []ProviderRewards
-	delegatePools []DelegatePool
-	desc          [][]string
+	rewards       map[string]currency.Coin
+	delegatePools map[string]map[string]currency.Coin
 }
 
-func aggregateProviderRewards(
-	spus []dbs.StakePoolReward, round int64,
-) (*providerRewardsDelegates, error) {
+func aggregateProviderRewards(spus []dbs.StakePoolReward) (*providerRewardsDelegates, error) {
 	var (
-		rewards       = make([]ProviderRewards, 0, len(spus))
-		delegatePools = make([]DelegatePool, 0, len(spus))
-		descs         = make([][]string, 0, len(spus))
+		rewardsMap   = make(map[string]currency.Coin)
+		dpRewardsMap = make(map[string]map[string]currency.Coin)
 	)
 	for i, sp := range spus {
 		if sp.Reward != 0 {
-			rewards = append(rewards, ProviderRewards{
-				ProviderID:                    sp.ProviderId,
-				Rewards:                       sp.Reward,
-				TotalRewards:                  sp.Reward,
-				RoundServiceChargeLastUpdated: round,
-			})
+			rewardsMap[sp.ProviderId] = rewardsMap[sp.ProviderId] + sp.Reward
 		}
-
-		// merge delegate rewards and penalties
-		for k, v := range spus[i].DelegateRewards {
-			delegatePools = append(delegatePools, DelegatePool{
-				ProviderID:           sp.ProviderId,
-				ProviderType:         sp.ProviderType,
-				PoolID:               k,
-				Reward:               v,
-				TotalReward:          v,
-				TotalPenalty:         spus[i].DelegatePenalties[k],
-				RoundPoolLastUpdated: round,
-			})
-		}
-
-		// append remaining penalties if any
-		for k, v := range spus[i].DelegatePenalties {
```

Comment:
Should we ignore this `DelegatePenalities`?

Reply:
I don't know. We should not be using delegate penalties, but they might still be used somewhere.

The problem with delegate penalties is that they can easily result in negative rewards. Negative rewards will never be collected and can be used to bypass paying fines.